Mathlib/Geometry/RingedSpace/Basic.lean

```diff
@@ -108,22 +116,34 @@ theorem isUnit_of_isUnit_germ (U : Opens X) (f : X.presheaf.obj (op U))
     rw [germ_res_apply, germ_res_apply]
     apply (h z ((iVU x).le hzVx)).mul_right_inj.mp
     -- Porting note: now need explicitly typing the rewrites
+    -- note: this is bad, I think we should replace the `FunLike` on
+    -- concrete category with `CoeFun`
+    rw [← CommRingCat.forget_map_apply]
     rw [← X.presheaf.germ_res_apply (iVU x) z hzVx f]
     -- Porting note: change was not necessary in Lean3
     change X.presheaf.germ _ z hzVx _ * (X.presheaf.germ _ z hzVx _) =
       X.presheaf.germ _ z hzVx _ * X.presheaf.germ _ z hzVy (g y)
+    rw [CommRingCat.forget_map_apply]
     rw [← RingHom.map_mul,
-      congr_arg (X.presheaf.germ (V x) z hzVx) (hg x),
-      X.presheaf.germ_res_apply _ _ _ f,
-      ← X.presheaf.germ_res_apply (iVU y) z hzVy f,
-      ← RingHom.map_mul,
-      congr_arg (X.presheaf.germ (V y) z hzVy) (hg y), RingHom.map_one, RingHom.map_one]
+      congr_arg (X.presheaf.germ (V x) z hzVx) (hg x)]
+    rw [← CommRingCat.forget_map_apply]
+    rw [← CommRingCat.forget_map_apply]
+    rw [← CommRingCat.forget_map_apply]
+    rw [X.presheaf.germ_res_apply _ _ _ f,
+      ← X.presheaf.germ_res_apply (iVU y) z hzVy f]
+    rw [CommRingCat.forget_map_apply, CommRingCat.forget_map_apply, CommRingCat.forget_map_apply]
+    rw [← RingHom.map_mul]
+    rw [congr_arg (X.presheaf.germ (V y) z hzVy) (hg y)]
+    rw [RingHom.map_one, RingHom.map_one]
```

Comment:
This is horrible. All this back and forth between `Hom.hom` and the `DFunLike.coe` coming from `ConcreteCategory.instFunLike` is needed to prevent the original `rw` from becoming an `erw`.

I think it would help if we replace `ConcreteCategory.instFunLike` by a `CoeFun`, but this of course also has some fallout, so I did not do a full test.